src/components/CreditCardBilling/CreditCardBilling.tsx

```diff
@@ -0,0 +1,299 @@
+    >
+      <canvas ref={chartCanvasRef} css={Canvas} />
+      <canvas ref={popoverCanvasRef} css={Canvas} />
+    </div>
+  );
+}
+
+const getCoordinatesAndPercentages = ({
+  data,
+  xMax,
+  yMax,
+  margin = defaultMargins,
+  padding = 4,
+}: {
+  data: CreditCardBilling[];
+  xMax: number;
+  yMax: number;
+  margin?: Sides;
+  padding?: number;
+}) => {
+  const withCoordinates: CreditCardBillingWithCoords[] = [];
+  const accumulativeValue = data.reduce((a, c) => {
+    return a + c.value;
+  }, 0);
+  const reRangedXMaxRatio =
+    (xMax - margin.right - margin.left) / accumulativeValue;
+  let rectStart = margin.left;
+
+  data.forEach((datum, i) => {
+    const isLast = i === data.length - 1;
+    const paddingAdjustment = isLast ? 0 : padding / 2;
+    const width =
+      accumulativeValue * (datum.value / accumulativeValue) * reRangedXMaxRatio;
+    const start = i === 0 ? rectStart : rectStart + padding / 2;
+    const end = rectStart + width - paddingAdjustment;
+    const percentage = parseFloat(
+      ((datum.value / accumulativeValue) * 100).toFixed(1)
+    );
+    withCoordinates.push({
+      ...datum,
+      x: { start, end },
+      y: {
+        start: margin.top,
+        end: yMax - margin.top - margin.bottom,
+      },
+      percentage,
+    });
+    rectStart = end + paddingAdjustment;
+  });
+
+  return withCoordinates;
+};
+
+const drawChart = ({
+  canvasRef,
+  containerRef,
+  data,
+  margin = defaultMargins,
+  padding = 4,
+  colorPallete = pallete,
+}: {
+  canvasRef: HTMLCanvasElement;
+  containerRef: HTMLDivElement;
+  data: Record<'data', CreditCardBilling[]>;
+  margin?: Sides;
+  padding?: number;
+  colorPallete?: Record<string, React.CSSProperties['color']>;
+}) => {
+  const containerDimension = containerRef.getBoundingClientRect();
+  const withCoordinates = getCoordinatesAndPercentages({
+    data: data.data.sort((a, b) => {
+      return b.value - a.value;
+    }),
+    xMax: containerDimension.width,
+    yMax: containerDimension.height,
+    padding,
+    margin,
+  });
+
+  const ctx = canvasRef.getContext('2d');
+  if (ctx === null) return;
+
+  withCoordinates.forEach((datum, i) => {
+    const { x, y } = datum;
+
+    if (i === 0 || i === withCoordinates.length - 1) {
+      ctx.fillStyle = `${colorPallete[datum.category]}`;
+      ctx.roundRect(x.start, y.start, x.end - x.start, y.end, [
+        i === 0 ? roundRectRadii : 0,
+        i === withCoordinates.length - 1 ? roundRectRadii : 0,
+        i === withCoordinates.length - 1 ? roundRectRadii : 0,
+        i === 0 ? roundRectRadii : 0,
+      ]);
+      ctx.fill();
+    } else {
+      ctx.fillStyle = `${pallete[datum.category]}`;
+      ctx.fillRect(x.start, y.start, x.end - x.start, y.end);
+    }
+  });
+};
+
+const selectDatum = ({
+  e,
+  containerRef,
+  data,
+  padding,
+  margin = defaultMargins,
+}: {
+  e: MouseEvent;
+  containerRef: HTMLDivElement;
+  data: Record<'data', CreditCardBilling[]>;
```

Comment:
이런 형태라면 어차피 키가 하나밖에 없는 맵이라 자료구조 상의 의의가 조금 떨어질 것 같슴다.

혹시 이렇게 바로 필요한 데이터를 넘기지 않는 이유가 있을까유??
```suggestion
  data: CreditCardBilling[];
```